flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1198,16 +1205,231 @@ class DidNotConverge(Exception):
+
+    :ivar test_case: The TestCase to use for assertions and cleanup callbacks.
+    :ivar deployer: The ``IDeployer`` that will be used to run changes.
+    :ivar cluster_state: The current cluster state, updated everytime
+        ``discover_state`` is run on the deployer.
+    :ivar UUID _dataset_id: The dataset_id of the dataset under test.
+    :ivar DesiredDataset inial_dataset: The first dataset state to converge to.
+    :ivar DesiredDataset next_dataset: The second dataset state to converge to.
+    """
+
+    def __init__(self, test_case, two_dataset_states, deployer=None):
+        """
+        Constructor for ``BlockDeviceCalculatorTestObjects``.
+
+        :param test_case: See instance variable ``test_case``.
+        :param two_datset_states: iterable of two ``DesiredDataset``s, such as
+            those generated from ``TWO_DESIRED_DATASET_STRATEGY``.
+        :param deployer: The ``IDeployer`` provider to use with the execution
+            of this test. Defaults to creating one using
+            ``create_blockdevicedeployer`` if none is specified.
+        """
+        if deployer is None:
+            deployer = create_blockdevicedeployer(test_case)
+        self.test_case = test_case
+        self.deployer = deployer
+        self.cluster_state = DeploymentState(
+            nodes=[_empty_node_state(deployer)]
+        )
+        self.dataset_id = two_dataset_states[0].dataset_id
+        self.initial_dataset, self.next_dataset = [
+            self._fixup_dataset_state(d)
+            for d in two_dataset_states
+        ]
+
+    def _fixup_dataset_state(self, dataset_state):
+        """
+        Modify a test dataset_state to a valid state for use with the given
+        deployer. Some parts of the sampled inputs created with the hypothesis
+        strategy are too fake for use with a real deployer. Currently this
+        entails:
+
+        - Set the mountpoint to a real mountpoint in desired dataset states
+          that have a mount point attribute.
+
+        :param DesiredDataset dataset_state: The state to fix up for this test
+            run.
+
+        :returns: A ``DesiredDataset`` modified for use in this test.
+        """
+        mount_point = self.deployer._mountpath_for_dataset_id(
+            unicode(self.dataset_id))
+        if getattr(dataset_state,
+                   'mount_point',
+                   _NoSuchThing) is not _NoSuchThing:
+            dataset_state = dataset_state.set(mount_point=mount_point)
+        return dataset_state
+
+    def current_local_state(self):
         """
-        Cleanup after running a hypothesis example.
+        Return the current ``BlockDeviceDeployerLocalState`` from the deployer.
+        """
+        local_state = self.test_case.successResultOf(
+            self.deployer.discover_state(_empty_node_state(self.deployer))
+        )
+
+        for change in local_state.shared_state_changes():
+            self.cluster_state = change.update_cluster_state(
```

Comment:
As discussed above, I don't think we actually need to inspect the `cluster_state` anyway, so this can probably just be removed.